slimta/envelope.py

```diff
@@ -24,19 +24,32 @@
 
 """
 
-from __future__ import absolute_import
+from __future__ import absolute_import, unicode_literals
 
 import re
 import copy
-import cStringIO
 from email.message import Message
 from email.generator import Generator
-from email.parser import Parser, FeedParser
+from email.parser import Parser
+
+import six
+from six.moves import cStringIO
+from six import BytesIO
```

Comment:
> This branch does not support Python 2.6 right? [...]

Ok, let's say that :)

> In 2.7, the BytesIO class is available in io.BytesIO just like 3.x, so it shouldn't need to be imported from six.

Ok, I'll do the switch.

> Also, what's the difference between cStringIO and BytesIO? In other files, you replaced occurrences of cStringIO with BytesIO, can that be done here too?

`cStringIO` handles unicode and `BytesIO` handles bytes (well, as all the time, with py2, `cStringIO` may handle both `unicode` or `bytes` a magic way).

As header data is handled internally as unicode data, it's cStringIO :-)